sql/core/src/test/scala/org/apache/spark/sql/Spark18601Suite.scala

```diff
@@ -0,0 +1,331 @@
+/*

```

Comment:
See the tests in that package for a example of how to create your own optimizer and stuff.